submissions/Harnytskyi/friends-app/js/script.js

```diff
@@ -0,0 +1,163 @@
+const GRID = document.getElementById("allFriends");
+const CHECKBOX_MAN = document.getElementById("gender-male");
+const CHECKBOX_WOMAN = document.getElementById("gender-female");
+const SORT_ALPHABET_ASCEND = document.getElementById("a-z");
+const SORT_ALPHABET_DESCEND = document.getElementById("z-a");
+const SEARCH_INPUT = document.getElementById('search');
+const SORT_AGE_ASCEND = document.getElementById("0-9");
+const SORT_AGE_DESCEND = document.getElementById("9-0");
+const FILTER_SORT = document.getElementById("filter-sort");
+const FILTER_GENDER = document.getElementById("filter-gender");
+let friends = [];
+let gender = {
+    male: true,
+    female: true
+}
+
+let selectedFriends = [...friends];
+function filterByGender() {
+    if (gender.male == true && gender.female == true)
+        selectedFriends = friends;
+    else if (gender.male == true && gender.female == false)
+        selectedFriends = friends.filter(item => item.gender == "male")
+    else if (gender.male == false && gender.female == true)
+        selectedFriends = friends.filter(item => item.gender == "female")
+    search();
+}
+function displayFriends() {
+    GRID.innerHTML = selectedFriends.map(item => {
+        return `<div id="${item.name}" class="card">
+            <img class="photo" src="${item.photo}">
+            <p class="name">${item.name}</p>
+            <p class="age">Age: ${item.age}</p>
+            <a class="phone" href="tel:${item.phone}">${item.phone}</a>
+            <div class="email">
+                <a href="mailto:${item.email}">
+                    <button class="button-message">SEND MESSAGE</button>
+                </a>
+            </div>
+        </div>`
+    }).join("");
+}
+
+function changeFilterStatus(checkbox) {
+    if (checkbox.checked) {
+        gender[checkbox.value] = true;
+    } else {
+        if (checkbox.value == "male") {
+            if (gender.female == false) {
+                gender.female = true;
+                CHECKBOX_WOMAN.click();
+            }
+        }
+        else if (checkbox.value == "female") {
+            if (gender.male == false) {
+                gender.male = true;
+                CHECKBOX_MAN.click();
+            }
+        }
+        gender[checkbox.value] = false;
+    }
+    filterByGender();
+    displayFriends();
+}
```

Comment:
My decision is that at least one option will always be shown. I decided to use it here because if the user disabling the checkboxes "man" and "woman" at the same time, he will be shown a blank page, which makes no sense.

As for the "custom clicking" remark, I just didn't guess the call necessary functionality directly. I will fix it

Reply:
now I have noticed that I can use two conditions in one `if`

Reply:
> My decision is that at least one option will always be shown. I decided to use it here because if the user disabling the checkboxes "man" and "woman" at the same time, he will be shown a blank page, which makes no sense.

Not sure what you mean, TBH :)

Reply:
It seems that I did not understand you because your code does not work completely. I propose the following solution:
```suggestion
function changeFilterStatus(checkbox) {
    if (!checkbox.checked){
        if (checkbox.value == "male" && gender.female == false) {
                CHECKBOX_WOMAN.checked = !CHECKBOX_WOMAN.checked;
                changeFilterStatus(CHECKBOX_WOMAN);
        }
        else if (checkbox.value == "female" && gender.male == false) {
                CHECKBOX_MAN.checked = !CHECKBOX_MAN.checked;
                changeFilterStatus(CHECKBOX_MAN);
        }
    }
    gender[checkbox.value] = !gender[checkbox.value];
    filterByGender();
    displayFriends();
}
```

Reply:
IMO, looks like overengineering. `changeFilterStatus(CHECKBOX_WOMAN);` - why do you need recursion here?

Also, when I click on `male` you will change the state of `female` checkbox? Wuuuut? :)

Reply:
There are only two categories ("male" and "female"). 
If the user does not search in one category, he must search in another.
To do this, I change the status of another checkbox.
![male-female-checkbox](https://user-images.githubusercontent.com/37344756/107644596-4ab63180-6c80-11eb-9e22-44b2bc91620e.gif)

Of course, I can make that no category be selected but I think this solution more interesting

Reply:
Ok, you have implemented your own custom controller for this. Now let's think from the user's perspective - is this controller behaves as expected or it's confusing? Usual behaviour for the checkbox is "when I click on `checkbox` it will change its own state". And not the state of the neighbour. So I believe it is confusing.

Also, you have two categories, but you anyway have 3 states:
1. female
1. male
1. all genders

And you can have "only one state selected".

So I see two ways:

1. Give the user a possibility to remove all checkboxes - which is bad for reasons you have mentioned.
1. Or implement controller which will show to the user that he can select 1 of 3 choices. Which is quite good for `radio` ;)

WDYT?

Reply:
Thank you for the meaningful explanation. 
You have convinced me that from a UX my decision is inexpedient. 
I decided to choose the first way and give a possibility to remove all checkboxes.